src/core/Akka.Remote/RemoteSystemDaemon.cs

```diff
@@ -304,45 +304,26 @@ private void HandleDaemonMsgCreate(DaemonMsgCreate message)
         /// <returns>ActorRef.</returns>
         public override IActorRef GetChild(IEnumerable<string> name)
         {
-            var elements = name.ToArray();
-            var path = elements.Join("/");
+            var path = name.Join("/");
             var n = 0;
             while (true)
             {
                 var nameAndUid = ActorCell.SplitNameAndUid(path);
-                var child = GetChild(nameAndUid.Name);
-                if (child == null)
+                if (TryGetChild(nameAndUid.Name, out var child))
                 {
-                    var last = path.LastIndexOf("/", StringComparison.Ordinal);
-                    if (last == -1)
+                    if (nameAndUid.Uid != ActorCell.UndefinedUid && nameAndUid.Uid != child.Path.Uid)
                         return Nobody.Instance;
-                    path = path.Substring(0, last);
-                    n++;
-                    continue;
+                    return n == 0 ? child : child.GetChild(name.TakeRight(n));
                 }
-                if (nameAndUid.Uid != ActorCell.UndefinedUid && nameAndUid.Uid != child.Path.Uid)
-                    return Nobody.Instance;
-
-                return n == 0 ? child : child.GetChild(elements.TakeRight(n));
-            }
-        }
-
 
-        private IInternalActorRef GetChild(string name)
```

Comment:
No need for extra `GetChild` method, which also calls `ActorCell.SplitNameAndUid` redundantly